aten/src/ATen/native/mkldnn/xpu/detail/Utils.cpp

```diff
@@ -349,4 +349,42 @@ bool binary_valid(
   return false;
 }
 
+bool is_channels_last(at::MemoryFormat fmt){
+  return (at::MemoryFormat::ChannelsLast == fmt) || (at::MemoryFormat::ChannelsLast3d == fmt);
+}
+
+bool is_smf_channels_last(const Tensor& t){
+  return is_channels_last(t.suggest_memory_format());
+}
+
+bool use_channels_last_for_conv(
+    const at::Tensor& src,
+    const at::Tensor& weight,
+    bool is_transpose){
+
+  if (!src.defined() || src.is_sparse()) {
+    // suggest channels_first
+    return false;
+  }
+
+  auto suggest_channels_last_format =
+      (is_smf_channels_last(src) || is_smf_channels_last(weight));
+  if (suggest_channels_last_format) {
+    // suggest channels_last
+    return true;
+  }
+
+  return false;
+}
+
+std::vector<int64_t> compatible_groups_deconv_strides(
+    const at::Tensor& weight,
+    dnnl::memory::dims group_size) {
+  std::vector<int64_t> strides = weight.strides().vec();
+  strides[0] = weight.strides()[1];
+  strides[1] = weight.strides()[0];
+  strides.insert(strides.begin(), group_size[2] * weight.strides()[0]);
+  return strides;
+}
+
```

Comment:
Those functions does not seep to be XPU specific (and I would be surprised if we don't have something like that for CPU already)
Can they be moved and we keep code that is specific to XPU

Reply:
hi @malfet, Great thanks for your comments. I have removed these small utils and make them inline in xpu files.